php-FIT-File-Analysis.php

```diff
@@ -792,17 +806,33 @@ private function fix_data($options) {
 			$this->interpolate_missing_data($missing_power_keys, $this->data_mesgs['record']['power']);
 		}
 	}
-	
+
 	/*
 	 * For the missing keys in the data, interpolate using values either side and insert as necessary.
 	 */
 	private function interpolate_missing_data(&$missing_keys, &$array){
+        if (!is_array($array)) {
+            // can't interpollate from non-array
+            return;
+        }
+
 		$num_points = 2;
 		$prev_value;
 		$next_value;
-		
+
+        $min_key = min(array_keys($array));
+        $max_key = max(array_keys($array));
+
 		for($i=0; $i<count($missing_keys); ++$i) {
 			if($missing_keys[$i] !== 0) {
+                // interpollating outside recorded range is impossible - use edge values instead
+                if ($missing_keys[$i] > $max_key) {
+                    $array[$missing_keys[$i]] = $array[$max_key];
+                    continue;
+                } else if ($missing_keys[$i] < $max_key) {
+                    $array[$missing_keys[$i]] = $array[$min_key];
+                    continue;
+                }
```

Comment:
When there are missing values before/after first/last valid value, it would ho into infinite loop